src/modules/powerrename/lib/PowerRenameManager.cpp

```diff
@@ -775,7 +775,38 @@ DWORD WINAPI CPowerRenameManager::s_regexWorkerThread(_In_ void* pv)
 
                                 // newName == nullptr likely means we have an empty search string.  We should leave newNameToUse
                                 // as nullptr so we clear the renamed column
-                                if (newName != nullptr)
+                                // Except string transformation is selected.
+                                
+                                if (newName == nullptr && (flags & Uppercase || flags & Lowercase || flags & Titlecase))
+                                {
+                                    newName = new wchar_t[MAX_PATH]{ 0 };
+                                    if (flags & NameOnly)
+                                    {
+                                        StringCchCopy(newName, MAX_PATH, fs::path(originalName).stem().c_str());
+                                    }
+                                    else if (flags & ExtensionOnly)
+                                    {
+                                        std::wstring extension = fs::path(originalName).extension().wstring();
+                                        if (!extension.empty())
+                                        {
+                                            if (extension.front() == '.')
+                                            {
+                                                extension = extension.erase(0, 1);
+                                            }
+                                            StringCchCopy(newName, MAX_PATH, extension.c_str());
+                                        }
+                                        else
+                                        {
+                                            newName = nullptr;
```

Comment:
Memory leak, add `delete[] newName;` before this line

Reply:
Also, at line 994, use `delete[] newName` instead of `CoTaskMemFree(newName);`

Reply:
I'm not sure about using `delete[]` here, since `newName` is allocated using `StrDupW` and from the documentation, it should be released using `LocalFree`:
https://docs.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-strdupw

Reply:
It's not because it's equal to `nullptr` at line 780.

Reply:
The problem here is that `newName` can be allocated by `LocalAlloc` or by `new` and that is not a good design, since one variable should not be allocated in different places with different allocators, otherwise the code maintenance will become problematic.

Reply:
Now the memory is allocated via StrDup(). SHStrDup() is used a lot in the project while StrDup() is used only once in PowerRenameRegEx.cpp:240 . I am not sure what is the difference between those two. Do you think using one of them is better than other?

Reply:
When using `SHStrDup`, the allocated string should be released using `CoTaskMemFree`.
When using `StrDup`, release it using `LocalFree`.
Other than that I don't know what are the differences, but please make sure to use the correct `free` method.

Reply:
Then we will have to change the original code too where in [PowerRenameRegEx.cpp:240](https://github.com/microsoft/PowerToys/blob/da38e6a23d525fca3f8fb1f7d3821d6dd97bc797/src/modules/powerrename/lib/PowerRenameRegEx.cpp#L236)  `StrDup` is used. Because the call to that function is ` spRenameRegEx->Replace(sourceName, &newName);` and then `newName` is freed by `CoTaskMemFree`. Right?

Reply:
Yes.